cpp/src/arrow/python/numpy_to_arrow.cc

```diff
@@ -424,27 +424,6 @@ inline Status NumPyConverter::ConvertData(std::shared_ptr<Buffer>* data) {
   return Status::OK();
 }
 
-template <>
-inline Status NumPyConverter::ConvertData<BooleanType>(std::shared_ptr<Buffer>* data) {
```

Comment:
Removing this to allow the casting kernel to do the conversion makes my test pass, but fails lots of others, so this is not the right fix.